### PR TITLE
global: clarify the API

### PR DESCRIPTION
**Pull request by jacquerie**
Closes https://github.com/inspirehep/inspire-dojson/issues/8, supersedes https://github.com/inspirehep/inspire-dojson/pull/157.

Users of this package typically want to do two things:

1. Given a string containing MARCXML, convert it to a JSON record.
2. Given a JSON record, convert it to a string containing MARCXML.

This commit makes the above tasks easier by introducing an API module
that offers two methods, ``marcxml2record`` and ``record2marcxml``,
which respectively do these two things.

Previously users had to import methods from the ``processors`` and the
``utils`` module, as well as a util from the upstream DoJSON package.
For example, code like

    >>> from dojson.contrib.marc21.utils import create_record
    >>> from inspire_dojson.processors import overdo_marc_dict
    >>> record = overdo_marc_dict(create_record(marcxml))

now becomes

    >>> from inspire_dojson import marcxml2record
    >>> record = marcxml2record(marcxml)

and code like

    >>> from inspire_dojson.hep import hep2marc
    >>> from inspire_dojson.utils import legacy_export_as_marc
    >>> marcxml = legacy_export_as_marc(hep2marc.do(record))

becomes

    >>> from inspire_dojson import record2marcxml
    >>> marcxml = record2marcxml(record)

Note in particular that, previously, users needed to know in advance
which kind of JSON records they were converting to MARCXML, while now
this is handled internally by ``record2marcxml``.

### CODE REVIEW

File: inspire_dojson/api.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        if _is_controlfield(tag, ind1, ind2):
+            value = force_single_element(values)
+            if not isinstance(value, text_type):
+                value = text_type(value)
+            record.append(CONTROLFIELD(value, {'tag': tag}))
+        else:
+            datafield = DATAFIELD({'tag': tag, 'ind1': ind1, 'ind2': ind2})
+            for value in force_list(values):
+                for code, el in sorted(iteritems(value)):
+                    if not isinstance(el, text_type):
```
Comment (michamos) [inline inspire_dojson/api.py]: is this handling repeated subfields, like multiple affiliations for one author?

Comment (jacquerie) [reply inspire_dojson/api.py]: Right, I didn't, so I added a test for this: https://github.com/inspirehep/inspire-dojson/pull/160/files#diff-625efbbf35a2290861f596fee1b19838R180.

**Comment by michamos** [comment]
Small nit, but why do you put the API in `.api`? I think you should consider reexporting the API from the `__init__.py`, as it's nicer to do `from inspire_dojson import record2marcxml` and is the way most other well-known python libraries do it (`requests`, `flask`, `setuptools`, etc.).

**Comment by jacquerie** [comment]
> Small nit, but why do you put the API in .api? I think you should consider reexporting the API from the __init__.py, as it's nicer to do from inspire_dojson import record2marcxml and is the way most other well-known python libraries do it (requests, flask, setuptools, etc.).

Good idea, I did it for `inspire-matcher`, so it makes sense to do it here too. Done!

**Pull request merged**